test/infrastructure/docker/controllers/dockermachine_controller.go

```diff
@@ -209,34 +232,48 @@ func (r *DockerMachineReconciler) reconcileNormal(ctx context.Context, machine *
 		dockerMachine.Status.LoadBalancerConfigured = true
 	}
 
+	// At, this stage, we are ready for bootstrap. However, if the BootstrapExecSucceededCondition is missing we add it and we
+	// requeue so the user can se the change of state before the bootstrap actually start.
+	if !conditions.Has(dockerMachine, infrav1.BootstrapExecSucceededCondition) {
```

Comment:
I don't think we should be relying on conditions from previous reconciliations here, since we are the controller setting the conditions. Can we switch to detecting this through introspection instead?

Reply:
Uhh... this can potentially apply to any info/flag a controller is setting in status, not only on conditions, and we are consistently relying on that information on our controllers.

However, in this specific case, this is specifically for ensuring a condition is reported to users as explained in a previous comment, so I find it weird to use introspection when what I want to check is explicit if a condition exists

Reply:
> Uhh... this can potentially apply to any info/flag a controller is setting in status, not only on conditions, and we are consistently relying on that information on our controllers.

We shouldn't be, unless it's a status value that we've previously set as part of the current reconciliation or a resource that is managed by another controller. We should likely file bugs for any places we are currently relying on status set on previous reconciliations.

Reply:
Yes, I believe this use case is fine and makes sense, thanks :)